pkcs8/src/version.rs

```diff
@@ -0,0 +1,53 @@
+use core::convert::TryFrom;
+
+use der::{Encodable, Encoder, Tagged};
+
+/// Version marker for PKCS#8 documents.
+///
+/// (RFC 5958 designates `0` and `1` as the only valid versions for PKCS#8 documents)
+#[derive(Clone, Debug, Copy)]
+pub enum Version {
+    /// Denotes PKCS#8 v1, used for [`crate::PrivateKeyInfo`] and [`crate::OneAsymmetricKey`]
+    V1 = 0,
+    /// Denotes PKCS#8 v2, only used for [`crate::OneAsymmetricKey`]
+    V2 = 1,
+}
+
+impl From<Version> for u8 {
+    fn from(version: Version) -> Self {
+        version as u8
+    }
+}
+
+impl TryFrom<der::Any<'_>> for Version {
+    type Error = der::Error;
+
+    fn try_from(any: der::Any<'_>) -> der::Result<Version> {
+        any.tag().assert_eq(Self::TAG)?;
+
+        match *any.as_bytes() {
+            [0x00] => Ok(Version::V1),
+            [0x01] => Ok(Version::V2),
+            _ => Err(der::ErrorKind::Value {
+                tag: der::Tag::Integer,
+            }
+            .into()),
+        }
+    }
+}
```

Comment:
How about something like this:

```suggestion
impl TryFrom<u8> for Version {
    type Error = Error;

    fn try_from(byte: u8) -> Result<Version, Error> {
        match byte {
            0 => Ok(Version::V1),
            1 => Ok(Version::V2),
            _ => Err(Error::Version(byte)),
        }
    }
}

impl<'a> TryFrom<der::Any<'a>> for Version {
    type Error = der::Error;

    fn try_from(any: der::Any<'a>) -> der::Result<Version> {
        u8::try_from(any)?.try_into().map_err(|_| {
            der::ErrorKind::Value {
                tag: der::Tag::Integer,
            }
            .into()
        })
    }
}
```

Reply:
`Error::Version(byte)`?

Reply:
Yeah, that'd be adding another variant to the `pkcs8::Error` struct. Or you can just do `Error::Decode` for now.

Reply:
I'll do Decode for now